tst/com/amazon/corretto/crypto/provider/test/EdDSATest.java

```diff
@@ -0,0 +1,281 @@
+    final KeyPair kp1 = nativeGen.generateKeyPair();
+    final KeyPair kp2 = nativeGen.generateKeyPair();
+
+    final byte[] pk1 = kp1.getPrivate().getEncoded();
+    final byte[] pk2 = kp2.getPrivate().getEncoded();
+
+    final byte[] pbk1 = kp1.getPublic().getEncoded();
+    final byte[] pbk2 = kp2.getPublic().getEncoded();
+
+    assertTrue(!Arrays.equals(pk1, pk2));
+    assertTrue(!Arrays.equals(pbk1, pbk2));
+  }
+
+  @Test
+  public void keyGenValidation() throws GeneralSecurityException {
+    // Generate Keys with ACCP & Sign/Verify with SunEC
+    final byte[] message = new byte[] {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    final KeyPair keyPair = nativeGen.generateKeyPair();
+
+    final PKCS8EncodedKeySpec privateKeyPkcs8 =
+        new PKCS8EncodedKeySpec(keyPair.getPrivate().getEncoded());
+    final X509EncodedKeySpec publicKeyX509 =
+        new X509EncodedKeySpec(keyPair.getPublic().getEncoded());
+
+    final KeyFactory kf = KeyFactory.getInstance("EdDSA", "SunEC");
+
+    final PrivateKey privateKey = kf.generatePrivate(privateKeyPkcs8);
+    final PublicKey publicKey = kf.generatePublic(publicKeyX509);
+
+    final Signature eddsa = Signature.getInstance("Ed25519", "SunEC");
+
+    eddsa.initSign(privateKey);
+    eddsa.update(message, 0, message.length);
+    final byte[] signature = eddsa.sign();
+    eddsa.initVerify(publicKey);
+    eddsa.update(message);
+    assertTrue(eddsa.verify(signature));
+  }
+
+  @Test
+  public void keyFactoryValidation() throws GeneralSecurityException {
+    final KeyPair keyPair = jceGen.generateKeyPair();
+
+    final byte[] privateKeyJCE = keyPair.getPrivate().getEncoded();
+    final byte[] publicKeyJCE = keyPair.getPublic().getEncoded();
+
+    final PKCS8EncodedKeySpec privateKeyPkcs8 = new PKCS8EncodedKeySpec(privateKeyJCE);
+    final X509EncodedKeySpec publicKeyX509 = new X509EncodedKeySpec(publicKeyJCE);
+
+    final KeyFactory kf = KeyFactory.getInstance("Ed25519", NATIVE_PROVIDER);
+
+    final byte[] privateKeyACCP = kf.generatePrivate(privateKeyPkcs8).getEncoded();
+    final byte[] publicKeyACCP = kf.generatePublic(publicKeyX509).getEncoded();
+
+    // Confirm that ACCP & SunEC keys are equivalent
+    assertTrue(Arrays.equals(privateKeyACCP, privateKeyJCE));
+    assertTrue(Arrays.equals(publicKeyACCP, publicKeyJCE));
+  }
+
+  @Test
+  public void jceInteropValidation() throws GeneralSecurityException {
+    // Generate keys with ACCP and use JCE KeyFactory to get equivalent JCE Keys
+    final KeyPair keyPair = nativeGen.generateKeyPair();
+
+    final PrivateKey privateKey = keyPair.getPrivate();
+    final PublicKey publicKey = keyPair.getPublic();
+
+    // Set up ACCP and JCE Signature Instances
+    final Signature nativeSig = Signature.getInstance("Ed25519", NATIVE_PROVIDER);
+    final Signature jceSig = Signature.getInstance("Ed25519", "SunEC");
+
+    // Sign with ACCP and verify with SunEC
+    final byte[] message = new byte[] {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
```

Comment:
Can we have more tests here? Maybe more messages and of different lengths: from 1 byte to 1024?

Reply:
Updated